src/RightsService.Business/Commands/RoleLocalization/EditRoleLocalizationCommand.cs

```diff
@@ -0,0 +1,98 @@
+using Microsoft.AspNetCore.JsonPatch.Operations;
+
+namespace LT.DigitalOffice.RightsService.Business.Commands.RoleLocalization
+{
+  public class EditRoleLocalizationCommand : IEditRoleLocalizationCommand
+  {
+    private readonly IAccessValidator _accessValidator;
+    private readonly IResponseCreater _responseCreator;
+    private readonly IRoleLocalizationRepository _roleLocalizationRepository;
+    private readonly IPatchDbRoleLocalizationMapper _roleLocalizationMapper;
+    private readonly IEditRoleLocalizationRequestValidator _validator;
+
+    public EditRoleLocalizationCommand(
+      IAccessValidator accessValidator,
+      IResponseCreater responseCreator,
+      IRoleLocalizationRepository roleLocalizationRepository,
+      IPatchDbRoleLocalizationMapper roleLocalizationMapper,
+      IEditRoleLocalizationRequestValidator validator)
+    {
+      _accessValidator = accessValidator;
+      _responseCreator = responseCreator;
+      _roleLocalizationRepository = roleLocalizationRepository;
+      _roleLocalizationMapper = roleLocalizationMapper;
+      _validator = validator;
+    }
+
+    public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid roleLocalizationId, JsonPatchDocument<EditRoleLocalizationRequest> request)
+    {
+      if (!await _accessValidator.IsAdminAsync())
+      {
+        return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.Forbidden);
+      }
+
+      if (!_validator.ValidateCustom(request, out List<string> errors))
+      {
+        return _responseCreator.CreateFailureResponse<bool>(
+          HttpStatusCode.BadRequest,
+          errors);
+      }
+
+      Operation<EditRoleLocalizationRequest> isActiveOperation = request.Operations.FirstOrDefault(
+        o => o.path.EndsWith(nameof(EditRoleLocalizationRequest.IsActive), StringComparison.OrdinalIgnoreCase));
+      Operation<EditRoleLocalizationRequest> nameOperation = request.Operations.FirstOrDefault(
+        o => o.path.EndsWith(nameof(EditRoleLocalizationRequest.Name), StringComparison.OrdinalIgnoreCase));
+
+      DbRoleLocalization roleLocalization = await _roleLocalizationRepository.GetAsync(roleLocalizationId);
+
+      if (isActiveOperation != default)
+      {
+        bool isActive = bool.Parse(isActiveOperation.value.ToString().Trim());
+
+        if (roleLocalization.IsActive == isActive)
+        {
+          return _responseCreator.CreateFailureResponse<bool>(
+            HttpStatusCode.BadRequest,
+            new List<string> { "Role localization already has this status." });
+        }
+
+        if (isActive
+          && await _roleLocalizationRepository.DoesLocaleExistAsync(roleLocalization.RoleId, roleLocalization.Locale))
+        {
+          return _responseCreator.CreateFailureResponse<bool>(
+            HttpStatusCode.BadRequest,
+            new List<string> { "Role must have only one localization per locale." });
+        }
+      }
+
+      if (nameOperation != default
```

Comment:
use validator

Reply:
DoesNameExistAsync(string locale, string name) requires roleLocalization's locale, can't get it in validator